main_pullrequest.go

```diff
@@ -70,18 +88,30 @@ func processGitHubPullRequest(
 					pr.Sender.GetLogin(),
 				)
 			}
-			err = startPRPipeline(log, prBranchName, pr, conf, isOrgMember)
+			err = retryOnError(retryParams{
+				retryFunc: func() error {
+					return startPRPipeline(log, prBranchName, pr, conf, isOrgMember)
+				},
+				compFunc: func(err_ error) bool {
+					re := regexp.MustCompile("Missing CI config file|" +
+						"No stages / jobs for this pipeline")
+					switch {
+					case err_ == nil:
+						return false
+					case re.MatchString(err_.Error()):
+						log.Infof("start pipeline for PR '%d' is skipped", pr.Number)
+						return true
+					case !re.MatchString(err_.Error()):
+						log.Errorf("failed to start pipeline for PR: %s", err)
+						return false
```

Comment:
Also this second case could very well be merged with the `default`.

Reply:
Good point, fixing!